Library/Homebrew/cli_parser.rb

```diff
@@ -0,0 +1,50 @@
+require "optparse"
+require "ostruct"
+
+module Homebrew
+  module CLI
+    class Parser
+      attr_accessor :parser
+
+      def initialize(&block)
+        @parser = OptionParser.new
+        @parsed_args = OpenStruct.new
+        instance_eval(&block)
+      end
+
+      def option(name, description: nil, type: nil, required: false)
+        handler_name = "handle_#{type}".to_sym
+        send(handler_name, name, description, required)
+      end
+
+      def handle_switch(name, description, _)
+        @parser.on(name, description) do
+          @parsed_args.send("#{option_to_name(name)}?=", true)
+        end
+      end
+
+      def handle_comma_array(name, description, _)
+        args = [name, "=MANDATORY", Array, description]
+        @parser.on(*args) do |list|
+          @parsed_args.send("#{option_to_name(name)}=", list)
+        end
+      end
+
+      def handle_flag(name, description, required)
+        args = [name, description, required ? "=MANDATORY" : "=[OPTIONAL]"]
```

Comment:
We can use `OptionParser::REQUIRED_ARGUMENT` and `OptionParser::OPTIONAL_ARGUMENT` here instead of these strings, which would be a bit nicer.